arch/arm64/src/imx9/imx9_lpspi.c

```diff
@@ -2084,8 +2068,106 @@ struct spi_dev_s *imx9_lpspibus_initialize(int bus)
 #endif
 
   leave_critical_section(flags);
-
   return (struct spi_dev_s *)priv;
 }
 
+/****************************************************************************
+ * Name: imx9_lpspibus_disable
+ *
+ * Description:
+ *   Disable selected bus unconditionally and set refcount to 0
+ *
+ * Input Parameters:
+ *   Port number (for hardware that has multiple SPI interfaces)
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void imx9_lpspibus_disable(int bus)
+{
+  switch (bus)
+  {
+#ifdef CONFIG_IMX9_LPSPI1
+    case 1:
+      modifyreg32(IMX9_LPSPI1_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi1dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI2
+    case 2:
+      modifyreg32(IMX9_LPSPI2_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi2dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI3
+    case 3:
+      modifyreg32(IMX9_LPSPI3_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi3dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI4
+    case 4:
+      modifyreg32(IMX9_LPSPI4_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi4dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI5
+    case 5:
+      modifyreg32(IMX9_LPSPI5_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi5dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI6
+    case 6:
+      modifyreg32(IMX9_LPSPI6_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi6dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI7
+    case 7:
+      modifyreg32(IMX9_LPSPI7_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi7dev.refcount = 0;
+      break;
+#endif
+#ifdef CONFIG_IMX9_LPSPI8
+    case 8:
+      modifyreg32(IMX9_LPSPI8_BASE + IMX9_LPSPI_CR_OFFSET, LPSPI_CR_MEN, 0);
+      g_lpspi8dev.refcount = 0;
+      break;
+#endif
+    default:
+      spierr("ERROR: Unsupported SPI bus: %d\n", bus);
+  }
+}
+
+/****************************************************************************
+ * Name: imx9_lpspibus_uninitialize
+ *
+ * Description:
+ *   Unitialize the selected SPI bus
+ *
+ * Input Parameters:
+ *   dev -      Device-specific state data
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void imx9_lpspi_uninitialize(struct spi_dev_s *dev)
+{
+  struct imx9_lpspidev_s  *priv = (struct imx9_lpspidev_s *)dev;
+
+  if (priv->refcount == 1)
```

Comment:
Why not just first decrease refcount (if refcount >  0) and then disable if refcount == 0 ?